### ISSUE TITLE
detect_to_retrieve: question about training detection model

### ISSUE THREAD

**Issue by zht022** [tensorflow/models#7223]
### System information
- **What is the top-level directory of the model you are using**: 
    /tensorflow/models/research/delf/delf/
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
    Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**:
    pip
- **TensorFlow version (use command below)**:
    1.14.0
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:
    cuda 9.0, cudnn 7
- **GPU model and memory**:
    1080ti, 11G memory

### Describe the problem
hi, @andrefaraujo , I'm reading the paper "Detect-to-Retrieve: Efficient Regional Aggregation for Image Search", and try to train faster-rcnn on google-landmark-bbox-dataset by myself. I'm wondering what the training config is? like number-of-gpus, learning-rate, any data-augmentation, and so on. Since I may use pytorch for detection, is it ok to use the same config as tensorflow? 

Another question is, when training/testing detection model, there are only 2 classes: landmark / no-landmark, every landmark belongs to same class, is it right?

I'm waiting for your reply with eager. Thanks.

**Comment by andrefaraujo** [comment]
We're using Resnet50-based Faster-RCNN. The model is initialized from an ImageNet pre-trained checkpoint. Here is the training configuration (based on [this](https://github.com/tensorflow/models/blob/master/research/object_detection/protos/train.proto) protocol buffer):

``
train_config: {
  num_steps: 250000
  batch_size: 20
  optimizer {
    momentum_optimizer: {
      learning_rate: {
        cosine_decay_learning_rate {
          learning_rate_base: 0.0003
          warmup_learning_rate: 0.00003
          warmup_steps: 5000
          total_steps: 250000
      }
      }
      momentum_optimizer_value: 0.9
    }
    use_moving_average: false
  }
  gradient_clipping_by_norm: 10.0
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
  max_number_of_boxes: 100
}
``


> Another question is, when training/testing detection model, there are only 2 classes: landmark / no-landmark, every landmark belongs to same class, is it right?

Correct

**Comment by rahuldhamecha** [comment]
Thanks @andrefaraujo 

1)
If there are only two classes in train and test then why does `detector.MakeDetector(sess, cmd_args.detector_path)` on line 160 in `extract_boxes.py` return 300 classes and scores of the respective classes? 

Am I missing something?

2)
I tried using the image of a cat for testing and the pre trained Fast RCNN created a bounding box around the cat. I am confused if this model is used for landmark detection or object detection?

I followed the instructions on this page `https://github.com/tensorflow/models/blob/master/research/delf/DETECTION.md`

**Comment by andrefaraujo** [comment]
1. Line 160 only constructs the detector function. The line that returns the boxes is 191. There is only one detectable class, so if you inspect the variable `class_indices_out`, you will notice that all of the values are `1`. (ie, all detected boxes are for the same class, landmark) So you are probably seeing 300 detections in different parts of the image, with different confidence scores.

2. I think the key here is to look at the predicted confidence score for the cat box. I'd expect it to be small, indicating it's not likely a landmark box. That being said, it's also possible that it would be doing something weird, since the training dataset does not contain many cats so it may not know how to behave properly in this case.

**Comment by rahuldhamecha** [comment]
Thank you @andrefaraujo

**Comment by zht022** [comment]
hi, @andrefaraujo, I have another question about detect_to_retreive, as you mentioned in the paper, you re-train DELF model on GLD, is there a validation set? If so, how to separate train/val data? Thanks

**Comment by andrefaraujo** [comment]
For training DELF on GLD (ie, the data [here](https://www.kaggle.com/google/google-landmarks-dataset)), we simply split the training set (`train.csv`) of GLD into 80% train and 20% validation sets (split is done per class).

**Comment by zht022** [comment]
> For training DELF on GLD (ie, the data [here](https://www.kaggle.com/google/google-landmarks-dataset)), we simply split the training set (`train.csv`) of GLD into 80% train and 20% validation sets (split is done per class).

Is there a certain split method, or just randomly split?

**Comment by andrefaraujo** [comment]
We just split it randomly.
